lib/dry/schema/message_compiler.rb

```diff
@@ -177,13 +180,16 @@ def lookup_options(arg_vals:, input:)
       end
 
       # @api private
-      def message_text(template, tokens, options)
-        text = template[template.data(tokens)]
+      def message_text(predicate, tokens, options)
+        result = messages.lookup(predicate, tokens, options)
 
-        return text unless full
+        return result if !result || !full
```

Comment:
N.B. this is a bit of a behavior change that I think makes sense—if there is no result, then return that lack of result. Previously, if there was no text, then a string with the rule name followed by a space would still be returned, which would be puzzling